src/DotNetWorker/Hosting/FunctionsApplicationBuilder.cs

```diff
@@ -0,0 +1,134 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Azure.Functions.Worker.Hosting;
+using Microsoft.Azure.Functions.Worker.Middleware;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Diagnostics.Metrics;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Extensions.Hosting;
+
+/// <summary>
+/// A builder for Azure Functions worker applications.
+/// </summary>
+public class FunctionsApplicationBuilder : IHostApplicationBuilder, IFunctionsWorkerApplicationBuilder
```

Comment:
Am I understanding correctly that one could use the config lambda parameter in 'ConfigureFunctionsWorkerDefaults` or the web variant, to basically get the inner builder here? Previously that was the only way, but now promoting this to this level, I guess we still have that way back into it too? Probably ultimately harmless, but maybe a touch confusing?

Reply:
There's no way to call `ConfigureFunctionsWorkerDefaults()` off of this. That hangs off of the `IHostBuilder`, which we're not exposing publicly here.

You could do something like:

```csharp
FunctionsApplicationBuilder funcBuilder = FunctionsApplication.CreateBuilder(args);
funcBuilder.Use<CustomMiddleware1>();

var innerBuilder = funcBuilder.Services.AddFunctionsWorkerDefaults();
innerBuilder.Use<CustomMiddleware2>();
```
Why would you? I dunno... :-)

The types are exposed in a variety of ways for composability, but ultimately it shouldn't matter how you call them.

Reply:
No, b/c you can't do this as it's not an `IHostBuilder`. It's an `IHostApplicationBuilder`.

```charp
builder.ConfigureFunctionsWebApplication(same_as_builder =>
{
    same_as_builder.UseMiddleware<ExceptionHandlingMiddleware>();
});
```

Maybe the confusing part is that I am going to be adding a new `ConfigureFunctionsWebApplication()` (which does not expose the callback) that is an extension off of `IHostApplicationBuilder`?: https://github.com/Azure/azure-functions-dotnet-worker/blob/brettsam/relayer/extensions/Worker.Extensions.Http.AspNetCore/src/FunctionsApplicationBuilderAspNetCoreExtensions.cs

Reply:
Ah, that explains it. Cool - not exposing the callback removes the potential for this confusion.